source/graphql/reflection.d

```diff
@@ -0,0 +1,84 @@
+module graphql.reflection;
+import graphql.traits;
+import std.traits;
+import std.meta;
+import vibe.data.json;
+
+package struct SchemaReflection(Schema)
+{
+	static SchemaReflection *instance() @safe
+	{
+		_instance.initialize();
+		return &_instance;
+	}
+
+	string[TypeInfo] classes;
+	string[][TypeInfo] derivatives;
+	string[][string] bases;
+	static struct TypeWithStrippedName
+	{
+		Json typeJson;
+		string name;
+		bool canonical;
+	}
+	TypeWithStrippedName[string] jsonTypes;
+
+
+	private:
+	static SchemaReflection _instance;
+
+	bool initialized;
+
+	static void builder(T)(ref SchemaReflection ths)
+	{
+		static if(is(T == class))
+		{
+			ths.classes[typeid(T)] = T.stringof;
+			static foreach(B; AliasSeq!(T, TransitiveBaseTypeTuple!T))
```

Comment:
try removing the static from the static foreach.
We found that static foreach takes a lot of memory.
And you don't need it here.

Reply:
OK